specs/mocks/functionsMocks.ts

```diff
@@ -0,0 +1,32 @@
+
+import { User } from '@prisma/client';
+import { user } from '../fixtures/users';
+import { bcryptInterface } from '../../src/interfaces/bcryptInterface';
+import { PrismaClientInterface } from '../../src/interfaces/prismaInterface';
+
+
+export const mockPrismaClient: PrismaClientInterface = {
+  user: {
+    findUnique: async ({ where: { email } }): Promise<User | null> => {
+      if (email === 'user@example.com') {
+        return Promise.resolve(user);
+      }
+      return null;
+    },
```

Comment:
we can change this to use a more expressive syntax. Let's use things like the implicit return of arrow functions and the ternary operator. 

- [Implicit return](https://medium.com/@bunlong/arrow-functions-return-rules-in-javascript-b63ed5f25994#:~:text=return%20value%3B%0A%7D-,2.%20Implicit%20Return,-What%20is%20Implicit)
- [Ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator)


```suggestion
    findUnique: async ({ where: { email } }): Promise<User | null> => 
      email === 'user@example.com' ? Promise.resolve(user) : null,
```